plugins/extend-router.js

```diff
@@ -19,6 +19,43 @@ VueRouter.prototype.applyQuery = function(qp, defaults = {}) {
   });
 };
 
+// https://github.com/vuejs/vue-router/issues/2881#issuecomment-520554378
```

Comment:
Do we need this? Just concerned about introducing this plugin at this stage - we should probably just add catches to the router.push/replace